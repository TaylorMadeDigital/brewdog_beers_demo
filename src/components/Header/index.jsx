import React from 'react';
import {
    Link
  } from "react-router-dom";
import logo from '../../assets/images/brewdog_logo.png';
import Heading from '../../components/Heading';
import { CONTENT } from '../../consts/content';

import s from './index.scss';

const Header = () => {
  return (
    <Link to="/">
        <header className={s.root}>
            <img src={logo} alt={CONTENT.global.logo} role="presentation" />
            <Heading headingLevel="h1">{CONTENT.global.title}</Heading>
        </header>
    </Link>
  );
}

export default Header;
