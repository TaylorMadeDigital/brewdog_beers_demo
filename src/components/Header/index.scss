@import '../../assets/scss/variables';

.root {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: $space-base;

  img {
    max-width: 75px;
    margin: 0 $space-base 0 0;
  }
}