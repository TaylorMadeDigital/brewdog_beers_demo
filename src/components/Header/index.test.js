import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, cleanup } from "@testing-library/react";
import Header from './index';

describe('<Header />', () => {

  afterEach(cleanup);

  it('snapshot: renders default state', () => {
    const { asFragment } = render(<Router><Header /></Router>);
    expect(asFragment()).toMatchSnapshot();
  });
});
