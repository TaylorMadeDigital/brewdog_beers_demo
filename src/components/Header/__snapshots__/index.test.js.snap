// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> snapshot: renders default state 1`] = `
<DocumentFragment>
  <a
    href="/"
  >
    <header>
      <img
        alt="Brewdog logo"
        role="presentation"
        src="brewdog_logo.png"
      />
      <h1
        class=""
        data-testid="heading"
      >
        Brewdog Beer Collection
      </h1>
    </header>
  </a>
</DocumentFragment>
`;
